### PR TITLE
Edit profile Bio

### PR DESCRIPTION
**Pull request by gguijarro-c-chwy**
Edit Profile Bio

image

### CODE REVIEW

File: lib/core/di/bloc_module.dart [hunk 1]
```diff
@@ -18,7 +18,14 @@ void _registerBlocsModule() {
         _getIt<AuthRepository>(),
       ));
   _registerFactory(() => BottomNavigationBloc());
-  _registerFactory(() => ProfileBloc(_getIt<FetchProfileUseCase>(), _getIt<HyphaSharedPrefs>()));
+  _registerFactory(() => ProfileBloc(
```
Comment (gguijarro-c-chwy) [inline lib/core/di/bloc_module.dart:21]: cleanup DI

File: lib/core/di/usecases_module.dart [hunk 2]
```diff
@@ -17,4 +17,7 @@ void _registerUseCasesModule() {
       _getIt<HyphaSharedPrefs>(),
     ),
   );
+  _registerFactory(() => SetNameUseCase(_getIt<AmplifyService>()));
```
Comment (gguijarro-c-chwy) [inline lib/core/di/usecases_module.dart:20]: cleanup DI

File: lib/ui/profile/components/profile_view.dart [hunk 3]
```diff
@@ -76,9 +78,19 @@ class ProfileView extends StatelessWidget {
                             child: HyphaActionableCard(
                               trailer: const Icon(Icons.edit),
                               onTap: () async {
-                                print('EDIT Tapped');
+                                showModalBottomSheet(
```
Comment (gguijarro-c-chwy) [inline lib/ui/profile/components/profile_view.dart:81]: show Bio enter sheet

File: lib/ui/profile/components/profile_view.dart [hunk 4]
```diff
@@ -141,7 +153,7 @@ class ProfileView extends StatelessWidget {
                               }
                               if (state.profileData != null) {
                                 final success = await as.loginUser(state.profileData!.account);
-                                print("login res: $success");
+                                print('login res: $success');
```
Comment (gguijarro-c-chwy) [inline lib/ui/profile/components/profile_view.dart:156]: cleanup

File: lib/ui/profile/interactor/profile_data.dart [hunk 5]
```diff
@@ -44,13 +44,24 @@ class ProfileData {
...
       avatar: avatar,
     );
   }
 
+  ProfileData updateBio(String? bio) => ProfileData(
+        name: name,
+        account: account,
+        bio: bio,
+        avatar: avatar,
+        avatarUrl: avatarUrl,
```
Comment (gguijarro-c-chwy) [inline lib/ui/profile/interactor/profile_data.dart:59]: @n13  do we need avatar and avatarUrl ?

File: lib/core/network/repository/auth_repository.dart [hunk 6]
```diff
@@ -51,35 +61,33 @@ class AuthRepository {
 
       _saveUserData(UserProfileData(accountName: accountName, userName: userName), userAuthData, false);
 
-      /// TODO(gguij) Improve this UX process.
+      // TODO(gguij): Improve this UX process.
```
Comment (gguijarro-c-chwy) [inline lib/core/network/repository/auth_repository.dart:64]: fix build

File: lib/ui/profile/interactor/profile_data.dart (hunk 5 above)
Comment (n13) [reply lib/ui/profile/interactor/profile_data.dart:59]: Yes

Comment (n13) [reply lib/ui/profile/interactor/profile_data.dart:59]: avatarURL is a resolved URL

avatar is the image name, like xyz.jpg

**Review by n13 — APPROVED** [review]

File: lib/ui/profile/interactor/profile_data.dart (hunk 5 above)
Comment (gguijarro-c-chwy) [reply lib/ui/profile/interactor/profile_data.dart:59]: and what is the avatar used for?

**Pull request merged**